src/rust/engine/process_execution/src/remote_cache_tests.rs

```diff
@@ -75,7 +75,9 @@ impl CommandRunnerTrait for MockLocalCommandRunner {
 // NB: We bundle these into a struct to ensure they share the same lifetime.
 struct StoreSetup {
   pub store: Store,
+  #[allow(dead_code)]
   pub store_dir: PathBuf,
+  #[allow(dead_code)]
```

Comment:
Is the idea that we should remove these in a followup? I can't imagine a sideeffect to deleting the `store_dir` if it is actually dead.

Reply:
Yeah, `store_dir` can definitely go :)

`cas` could probably do with a comment explaining that it's being kept alive for side-effects too

Reply:
Should I rename `cas` to `_cas` then instead of using `#[allow(dead_code)]`?

Reply:
Also, `store_dir` comes from:
```
let store_dir = TempDir::new().unwrap().path().join("store_dir");`
```

`TempDir` is just a temporary in that expression and gets dropped immediately, which means `store_dir` is made and then deleted. We should store the `TempDir` in the struct and not the `PathBuf`.

Reply:
I'll fix in a separate PR.

Reply:
https://github.com/pantsbuild/pants/pull/12464